libdrgn/program.c

```diff
@@ -207,12 +208,21 @@ drgn_program_check_initialized(struct drgn_program *prog)
 static struct drgn_error *has_kdump_signature(const char *path, int fd,
 					      bool *ret)
 {
-	char signature[KDUMP_SIG_LEN];
+	char signature[KDUMP_SIG_READ_LEN];
```

Comment:
Instead of defining `KDUMP_SIG_READ_LEN`, you can do this here, which makes the intention clear:

```suggestion
	char signature[max_iconst(KDUMP_SIG_LEN, FLATTENED_SIG_LEN)];
```

Reply:
Thank you, I wasn't sure what the proper "max()" definition was and I knew redefining it in the header wouldn't be good so I fudged it.